lmfit/models.py

```diff
@@ -735,6 +735,60 @@ def guess(self, data, x, negative=False, **kwargs):
     guess.__doc__ = COMMON_GUESS_DOC
 
 
+class Pearson4Model(Model):
+    r"""A model based on a Pearson IV distribution.
+
+    The model has five parameters: `amplitude` (:math:`A`), `center`
+    (:math:`\mu`), `sigma` (:math:`\sigma`), `expon` (:math:`m`) and `skew` (:math:`\nu`).
+    In addition, parameters `fwhmapprox`, `height` and `position` are included as
+    constraints to report estimates for the approximate full width at half maximum, the
+    peak height, and the peak position, respectively.
+    The fwhmapprox value has an error of about 20% in the
+    parameter range expon: (0.5, 1000], skew: [-1000, 1000].
+
+    .. math::
+
+        f(x;A,\mu,\sigma,m,\nu)=A \frac{\left|\frac{\Gamma(m+i\tfrac{\nu}{2})}{\Gamma(m)}\right|^2}{\sigma\beta(m-\tfrac{1}{2},\tfrac{1}{2})}\left[1+\frac{(x-\mu)^2}{\sigma^2}\right]^{-m}\exp\left(-\nu \arctan\left(\frac{x-\mu}{\sigma}\right)\right)
+
+    where :math:`\beta` is the beta function (see :scipydoc:`special.beta`).
+    The :meth:`guess` function always gives a starting value of 1.5 for `expon`,
+    and 0 for `skew`. In addition, parameters `fwhmapprox`, `height` and `position` are included as
+    constraints to report approximate full width at half maximum, peak height, and peak position, respectively.
+
+    For more information, see:
+    https://en.wikipedia.org/wiki/Pearson_distribution#The_Pearson_type_IV_distribution
+
+    """
+
+    def __init__(self, independent_vars=['x'], prefix='', nan_policy='raise',
+                 **kwargs):
+        kwargs.update({'prefix': prefix, 'nan_policy': nan_policy,
+                       'independent_vars': independent_vars})
+        super().__init__(pearson4, **kwargs)
+        self._set_paramhints_prefix()
+
+    def _set_paramhints_prefix(self):
+        self.set_param_hint('expon', value=1.5, min=0.5+tiny, max=1000)
+        self.set_param_hint('skew', value=0.0, min=-1000, max=1000)
+        fmt = ("{prefix:s}sigma*sqrt(2**(1/{prefix:s}expon)-1)*pi/arctan2(exp(1)*{prefix:s}expon, {prefix:s}skew)")
+        self.set_param_hint('fwhmapprox', expr=fmt.format(prefix=self.prefix))
+        fmt = ("({prefix:s}amplitude / {prefix:s}sigma) * exp(2 * (real(loggammafcn({prefix:s}expon + {prefix:s}skew * 0.5j)) - loggammafcn({prefix:s}expon)) - betalnfnc({prefix:s}expon-0.5, 0.5) - "
+               "{prefix:s}expon * log1p(square({prefix:s}skew/(2*{prefix:s}expon))) - {prefix:s}skew * arctan(-{prefix:s}skew/(2*{prefix:s}expon)))")
+        self.set_param_hint('height', expr=fmt.format(tiny, prefix=self.prefix))
+        fmt = ("{prefix:s}center-{prefix:s}sigma*{prefix:s}skew/(2*{prefix:s}expon)")
+        self.set_param_hint('position', expr=fmt.format(prefix=self.prefix))
```

Comment:
out of curiosity, where do those forms for "fwhm" and "position" come from?  I'm sure we'll get asked at some point.   And what does "position" mean? Is that meant to be "mean position", "position at highest value", or something else?  Those might be nice to have in docstrings.

Reply:
Hi,

the `position` is meant to be the position of the highest value (if you put `position` as `x` in pearson4, you would get the `height` value). The formula for position could be easily derived by taking the first derivative and solving it for 0.

I did not find a formula for the `fwhm` value in literature, thus I had to figure out an own one. The approximation is based on the formula for pearson7, but of course with modifications taking the contribution of skew into account. I keep searching for a more accurate one!

Reply:
@lellid  Thanks.  
For `position`:  OK, maybe the docstring could be a little more explicit about this.   There are a few other lineshapes where `center` might not be the x value of the maximum height.  I agree that in those cases, knowing both are good.   It's just that in my view, you are asking *us* to support (including, answer questions) about this function.   I just want it to be a little clearer because I will definitely forget if it is not documented. ;).

For  `fwhm`:  if approximate is all we have, that's OK.  Maybe describe it as "approximate" in the docstring.